File: frontend/src/types/events.ts
export type GameEventApiData = {
    id: number;
    province_id: string | null;
    province_name?: string | null;
    region_name?: string | null;
    event_type: string;
    actor_user_id: number;
    actor_username: string;
    actor_faction_id?: number | null;
    previous_owner_user_id: number | null;
    previous_owner_username: string | null;
    problem_title_slug: string | null;
    problem_title: string | null;
    problem_difficulty: string | null;
    points: number | null;
    runtime_ms: number | null;
    previous_runtime_ms: number | null;
    created_at: string;
};

File: backend/app/schemas/events.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, ConfigDict


class MapEventResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    friendship_id: int
    weekly_map_id: int
    province_id: str
    event_type: str
    actor_user_id: int
    actor_username: str
    previous_owner_user_id: Optional[int] = None
    previous_owner_username: Optional[str] = None
    problem_title_slug: str
    problem_title: Optional[str] = None
    problem_difficulty: Optional[str] = None
    points: Optional[int] = None
    runtime_ms: Optional[int] = None
    previous_runtime_ms: Optional[int] = None
    created_at: datetime


class FeedItemResponse(MapEventResponse):
    friend_id: int
    friend_username: str
